sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -303,94 +303,44 @@ case class LoadDataCommand(
-          val pathPattern = new File(dirPath.toAbsolutePath.toString, file.getName).toURI.getPath
-          val safePathPattern = if (Utils.isWindows) {
-            // On Windows, the pattern should not start with slashes for absolute file paths.
-            pathPattern.stripPrefix("/")
-          } else {
-            pathPattern
-          }
-          val files = new File(dir).listFiles()
-          if (files == null) {
-            false
-          } else {
-            val matcher = fileSystem.getPathMatcher("glob:" + safePathPattern)
-            files.exists(f => matcher.matches(fileSystem.getPath(f.getAbsolutePath)))
-          }
-        } else {
-          new File(file.getAbsolutePath).exists()
-        }
-        if (!exists) {
-          throw new AnalysisException(s"LOAD DATA input path does not exist: $path")
-        }
-        uri
+        val localFS = FileContext.getLocalFSFileContext()
+        makeQualified(FsConstants.LOCAL_FS_URI, localFS.getWorkingDirectory(), new Path(path))
       } else {
-        val uri = new URI(path)
-        val hdfsUri = if (uri.getScheme() != null && uri.getAuthority() != null) {
-          uri
-        } else {
-          // Follow Hive's behavior:
-          // If no schema or authority is provided with non-local inpath,
-          // we will use hadoop configuration "fs.defaultFS".
-          val defaultFSConf = sparkSession.sessionState.newHadoopConf().get("fs.defaultFS")
-          val defaultFS = if (defaultFSConf == null) {
-            new URI("")
-          } else {
-            new URI(defaultFSConf)
-          }
-
-          val scheme = if (uri.getScheme() != null) {
-            uri.getScheme()
-          } else {
-            defaultFS.getScheme()
-          }
-          val authority = if (uri.getAuthority() != null) {
-            uri.getAuthority()
-          } else {
-            defaultFS.getAuthority()
-          }
-
-          if (scheme == null) {
-            throw new AnalysisException(
-              s"LOAD DATA: URI scheme is required for non-local input paths: '$path'")
-          }
-
-          // Follow Hive's behavior:
-          // If LOCAL is not specified, and the path is relative,
-          // then the path is interpreted relative to "/user/<username>"
-          val uriPath = uri.getPath()
-          val absolutePath = if (uriPath != null && uriPath.startsWith("/")) {
-            uriPath
-          } else {
-            s"/user/${System.getProperty("user.name")}/$uriPath"
-          }
-          new URI(scheme, authority, absolutePath, uri.getQuery(), uri.getFragment())
-        }
-        val hadoopConf = sparkSession.sessionState.newHadoopConf()
-        val srcPath = new Path(hdfsUri)
-        val fs = srcPath.getFileSystem(hadoopConf)
-        if (!fs.exists(srcPath)) {
-          throw new AnalysisException(s"LOAD DATA input path does not exist: $path")
-        }
-        hdfsUri
+        val loadPath = new Path(path)
```

Comment:
The change of this line is a behavior change. @sujith71955 

We need to add this to the migration guide.

Reply:
you mean the exception which we are throwing now is different, we need to mention the same in migration guide also.
actually now user can also apply wildcard in the local folder path- shall we need to add this in guide

Reply:
I added a quick summary of the behavior change at https://issues.apache.org/jira/browse/SPARK-23425 in the Docs Text field. Feel free to add more. Is it more than the fact that certain characters are now wildcards?

Reply:
```Scala
val p = new Path("a b")
val p2 = new Path(new URI("a%20b"))
p2 == p
```

Reply:
Ah right. I think it's necessary to not parse it as a URI because it could contain "?" and that's a reserved char in URIs. We don't want any other behavior to change though; paths with a space in it should still work for example. @sujith71955 do you have maybe a quick test for a case or two like that? and/or @gatorsmile do you know of a case where user-visible behavior has changed (aside from wildcards)?

Reply:
@gatorsmile @srowen I tested the scenario by providing the spaces in the path, attaching the snapshots and observations
![image](https://user-images.githubusercontent.com/12999161/45349239-5f350200-b5ce-11e8-817b-10c4809ad23e.png)
![image](https://user-images.githubusercontent.com/12999161/45358119-0bceae00-b5e6-11e8-8ef5-10814987ea02.png)

Reply:
I will update the same in Migration guide.

Reply:
Did a small update in the summary, at https://issues.apache.org/jira/browse/SPARK-23425,  let me know for any suggestions

Reply:
Would loading '/tmp/dir1/test\ file.csv' succeed before this change?
What about adding a quick test for a case like this too?

Reply:
Nope.... we need to give %20 as space representation in older version
![image](https://user-images.githubusercontent.com/12999161/45367012-e057bd80-b5fd-11e8-8956-896e31c2bb78.png)

Reply:
if we give %20 it will work fine..  but this is expected behavior as per our current code.

Reply:
i mean in older versions compare to the latest

Reply:
gatorsmile is mentioning about this change to be mentioned in Migration guide

Reply:
Let's break this down:

Spark before change:
- "test\ file.csv" does not work
- "test%20file.csv" works? or fails to load? I see two statements here

Spark after change:
- "test\ file.csv" works
- "test%20file.csv" does not work

What is Hive's behavior? I assume it's the latter?

If so, then I think this change is properly viewed as a bug fix. But if "test%20file.csv" succeeded before, it's more complicated: inconsistent with Hive, but worked.

Reply:
Spark before change:

"test file.csv" does not work - yes 
"test%20file.csv" works? or fails to load? I see two statements here
 load command successful but no data is shown while querying which means the use-case is not working properly

Spark after change:

"test file.csv" works - Yes 
"test%20file.csv" does not work - Yes

Hive behaviour
"test file.csv" works - Yes 
"test%20file.csv" does not work - Yes
Our new changes are inline with Hive.